model/env.go

```diff
@@ -37,6 +42,64 @@ func (em *EnvVarMap) Validate() error {
 	return nil
 }
 
+// ClearOrPatch takes a new EnvVarMap and patches changes into the existing
+// EnvVarMap with the following logic:
+//  - If the new EnvVarMap is empty, clear the existing EnvVarMap completely.
+//  - If the new EnvVarMap is not empty, apply normal patch logic.
+func (em *EnvVarMap) ClearOrPatch(new *EnvVarMap) bool {
```

Comment:
since the input is a pointer, does it need to be checked that it's not `nil` before we try to dereference it on line 50?
```
gore> %                                                                                             ~ » gore
gore version 0.5.0  :help for help
gore> type MyMap map[string]string
gore> var m *MyMap
gore> m == nil
true
gore> len(*m)
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4cd87f]

goroutine 1 [running]:
main.main()
+0x1f
exit status 2
gore> 
```

Reply:
Thanks for the input. I wasn't originally checking for nil there because I knew the caller would be checking, but by handling the nil check in this method I can clean up the code in the caller so looks like a win/win to me!